libs/common/src/database/service.abstract.ts

```diff
@@ -0,0 +1,117 @@
+import {
+    BadRequestException,
+    ConflictException,
+    HttpException,
+    HttpStatus,
+    NotFoundException,
+} from '@nestjs/common';
+import {
+    DeepPartial,
+    FindOneOptions,
+    QueryRunner,
+    Repository,
+    SelectQueryBuilder,
+} from 'typeorm';
+import { Criteria, FindOptions, SortParam } from './model/database.model';
+import { isArray } from 'lodash';
+
+export class TableMetadata {
+    name: string;
+    tableName: string;
+    searchName: string;
+}
+
+export abstract class AbstractService<TEntity> {
+    name: string;
+    tableName: string;
+    searchName: string;
+
+    constructor(
+        private repository: Repository<TEntity>,
+        private queryBuilder: SelectQueryBuilder<TEntity>,
+        private queryRunner: QueryRunner,
+        metadata: TableMetadata,
+    ) {
+        this.name = metadata.name;
+        this.tableName = metadata.tableName;
+        this.searchName = metadata.searchName;
+    }
+
+    async finOne(findOptions: FindOneOptions<TEntity>) {
+        const entity = await this.repository.findOne(findOptions);
+
+        if (!entity) throw new NotFoundException(`${this.name} not found.`);
+
+        return {
+            [`${this.name.toLowerCase()}`]: entity,
+        };
+    }
+
+    async create(dto: DeepPartial<TEntity>) {
+        const entity = this.repository.create(dto);
+
+        try {
+            await this.repository.save(entity);
+
+            return {
+                [`${this.name.toLowerCase()}`]: entity,
+            };
```

Comment:
Seria legal você colocar o retorno do save em uma variável e retorná-la uma vez que o typeorm após o save coloca todos os atributos(id, createdAt, updatedAt, etc) da entidade:

```suggestion
        const entity = this.repository.create(dto);

        try {
            const createdEntity = await this.repository.save(entity);

            return {
                [`${this.name.toLowerCase()}`]: createdEntity,
            };
```